app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateFromURLController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateFromURLController', function ($scope, $routeParams, $location, $filter, AuthService, DataService, AlertMessageService, Navigate, ProjectsService ) {
+    AuthService.withUser();
+
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    $scope.alerts = {};
+    $scope.selected = {};
+
+    var alertInaccessibleNamespace = function(namespace, e) {
+      $scope.alerts.inaccessibleNamespace = {
+        type: "error",
+        message: "You do not have access to the namespace \"" + namespace + "\".",
+        details: "Reason: " + $filter('getErrorDetails')(e)
+      };
+    };
+
+    var alertInvalidImageStream = function(imageStream, e) {
+      $scope.alerts.invalidImageStream = {
+        type: "error",
+        message: "The requested image stream \"" + imageStream + "\" is not valid.",
```

Comment:
Suggest a more general message saying "Could not load image stream ..." instead of saying "not valid." It might be valid, but we couldn't load it for another reason. Same for templates.